controllers/postController.js

```diff
@@ -132,26 +153,50 @@ router.post('/:id/comment', verifyToken, (req, res) => {
  * @param {string} cid - comment's id
  */
 router.put('/:pid/comment/:cid', verifyToken, (req, res) => {
-  Post.updateComment(req.params.pid, req.params.cid, req.body).then(comment => {
-    res.json(comment);
-  })
-  .catch(err => {
+  Post.findOne({_id: req.params.pid})
+  .then(post => {
+    const userIsCommentAuthor = post.comments.find(obj => {return obj.author == req.user._id});
```

Comment:
This would seem to allow someone to edit any comment for a post that they have, at one point, commented on.. ie if me and Bob comment on a post, and I try to edit Bob's comment, this logic would allow me to even though it's not my comment (since it searches all post comments). We should be able to compare `obj._id` and `req.params.cid` to match the specific comment.

Reply:
Nice catch Neil 😄